restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentSearchPageState.java

```diff
@@ -0,0 +1,41 @@
+package io.stargate.web.docsapi.dao;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.nio.ByteBuffer;
+import java.util.Base64;
+
+public class DocumentSearchPageState {
+  @JsonProperty("documentId")
+  String documentId;
+
+  @JsonProperty("internalPageState")
+  String internalPageState;
+
+  @JsonCreator
+  public DocumentSearchPageState(
+      @JsonProperty("documentId") final String documentId,
+      @JsonProperty("internalPageState") final String internalPageState) {
+    this.documentId = documentId;
+    this.internalPageState = internalPageState;
+  }
+
+  public DocumentSearchPageState(final String documentId, final ByteBuffer internalPageStateBuf) {
+    this.documentId = documentId;
+    this.internalPageState = Base64.getEncoder().encodeToString(internalPageStateBuf.array());
```

Comment:
```suggestion
    this.documentId = Strings.nullToEmpty(documentId);
    if (internalPageStateBuf != null) {
    this.internalPageState = Base64.getEncoder().encodeToString(internalPageStateBuf.array());
    }
    else {
         this.internalPageState = ""
    }
```

Reply:
If `internalPageStateBuf` is `null`, it's probably a processing mistake on the server side. I'd rather throw an `IllegalArgumentException` in that case. WDYT?

Reply:
I think that's probably best, yes - I'm going to be making all of these changes on the branch I'm currently working through btw, since it's based off this branch: https://github.com/stargate/stargate/pull/748

Reply:
Oh actually it is possible for a valid DocumentSearchPageState to have an internalPageStateBuf that is `null`, since `null` means "the very start of the results"

Say you have 5 small docs that fit in a single C* page; if a user requests just one doc, then the paging state that is given back to them could be represented by `{"documentId": 1, "internalPageState": ""}` (essentially: I last saw document 1 on the first page of results) - and then when that's. sent back over the wire `internalPageStateBuf` is null, and that null is sent down to C*